host-ext/crouton/background.js

```diff
@@ -404,6 +411,53 @@ function websocketMessage(evt) {
         }
 
         break;
+    case 'N': /* Raise a notification */
+        /* Payload in JSON format, compatible with chrome.extensions specifications */
+        try {
+            var data = JSON.parse(payload);
+
+            if (!data.type)
+                data.type = "basic";
+            if (!data.iconUrl)
+                data.iconUrl = "icon-128.png";
```

Comment:
Could we take it one step further and provide a `-i` parameter which automatically encodes the image for you (switch id to `-I` since it'll be much less commonly used)?
base64 is easy (and is already a dependency of the extension), and it looks like Chrome figures out what type of file it is regardless of the explicit MIME type (so data:image/png should work even if it's a jpg).